applications/nrf_desktop/doc/motion.rst

```diff
@@ -141,12 +156,34 @@ Upon connection, the following happens:
 
     a. Switches to ``STATE_FETCHING``.
     #. Samples the motion data.
-    #. Submits the ``motion_event``.
-    #. Waits for the indication that the ``motion_event`` data was transmitted to the host.
-       This is done when the module receives the ``hid_report_sent_event`` event.
+    #. Submits the :c:struct:`motion_event`.
+    #. Waits for the indication that the :c:struct:`motion_event` data was transmitted to the host.
+       This is done when the module receives the :c:struct:`hid_report_sent_event` event.
 
-#. At that point, a next motion sampling is performed and the next ``motion_event`` sent.
+#. At that point, a next motion sampling is performed and the next :c:struct:`motion_event` sent.
 
 The module continues to sample data until disconnection or when there is no motion detected.
 The ``motion`` module assumes no motion when a number of consecutive samples equal to :ref:`CONFIG_DESKTOP_MOTION_SENSOR_EMPTY_SAMPLES_COUNT <config_desktop_app_options>` returns zero on both axis.
 In such case, the module will switch back to ``STATE_IDLE`` and wait for the motion sensor trigger.
+
+Simulated movement data
+=======================
+
+Current position in the generated trajectory is determined by a hardware cycle counter.
+If available, the module uses :c:func:`sys_clock_cycle_get_64` function to read the hardware cycle counter when generating motion.
+Otherwise the :c:func:`sys_clock_cycle_get_32` function is used for that purpose.
+
+When a HID subscriber connects, that is when device is connected either over USB or Bluetooth LE, the module submits :c:struct:`motion_event` containing relative motion calculated from the previously reported position.
+When the previously generated motion data is sent to the subscriber, that is when :c:struct:`hid_report_sent_event` is received by the module, the simulated motion module submits a subsequent event with position generated from updated timestamp.
+
+Accuracy of the generated movement data
+---------------------------------------
+
+The accuracy of generated simulated movement data depends on accuracy of the available hardware cycle counter.
+If the module is polled by the host with big frequency, that is when the :c:struct:`hid_report_sent_event` is received often by the module, the generated :c:struct:`motion_event` may contain value of ``0`` for both X and Y axes.
+
+.. note::
+   A HID report with reporting motion of ``0`` for both axes may be ignored by host computer's tools analyzing the HID report rate.
+
+   Increase value of the used scale factor or reduce time for transition between two points in the trajectory to increase the generated motion values.
```

Comment:
add reference to KConfig option for the scale factor

Reply:
Added a generic note that links to module's configuration section (it seems more suitable after I applied reword)